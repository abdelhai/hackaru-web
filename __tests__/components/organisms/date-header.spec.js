import MockDate from 'mockdate';
import Factory from '@/__tests__/__setups__/factory';
import DateHeader, { periods } from '@/components/organisms/date-header';
import { parse } from 'date-fns';

describe('DateHeader', () => {
  let factory;
  let wrapper;

  MockDate.set('2019-01-31T01:23:45');

  beforeEach(() => {
    factory = new Factory(DateHeader, {
      stubs: ['v-date-picker', 'no-ssr'],
      propsData: {
        date: '2018-01-01',
        periods: [periods.day, periods.week, periods.month, periods.year]
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when does not has today', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setProps({ periodIndex: 0 });
    });

    it('show today button', () => {
      expect(wrapper.contains('.today-button')).toBe(true);
    });
  });

  describe('when has today', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setProps({ periodIndex: 0, date: new Date() });
    });

    it('hide today button', () => {
      expect(wrapper.contains('.today-button')).toBe(false);
    });
  });

  describe('when click left arrow button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find('.left-arrow-button').vm.$emit('click');
    });

    it('emit left', () => {
      expect(wrapper.emitted('left')).toBeTruthy();
    });
  });

  describe('when click right arrow button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find('.right-arrow-button').vm.$emit('click');
    });

    it('emit right', () => {
      expect(wrapper.emitted('right')).toBeTruthy();
    });
  });

  describe('when click today button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find({ ref: 'today-button' }).vm.$emit('click');
    });

    it('emit update:date with today date', () => {
      expect(wrapper.emitted('update:date')[0]).toEqual([parse(Date.now())]);
    });
  });

  describe('when change selected period', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find({ ref: 'period-select' }).vm.$emit('input', '1');
    });

    it('emit update:periodIndex', () => {
      expect(wrapper.emitted('update:periodIndex')[0]).toEqual([1]);
    });
  });
});
