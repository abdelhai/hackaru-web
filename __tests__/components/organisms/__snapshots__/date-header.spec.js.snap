// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateHeader render correctly 1`] = `
<content-header-stub>
  <div
    class="date-heading"
  >
    <btn-stub
      aria-label="ariaLabels.previous"
      class="left-arrow-button has-icon"
    >
      <icon-stub
        class="is-primary is-large"
        name="chevron-left-icon"
      />
    </btn-stub>
     
    <no-ssr-stub>
      <v-date-picker-stub
        formats="[object Object]"
        show-caps=""
        value="Mon Jan 01 2018 00:00:00 GMT+0900 (JST)"
      />
    </no-ssr-stub>
     
    <btn-stub
      aria-label="ariaLabels.next"
      class="right-arrow-button has-icon"
    >
      <icon-stub
        class="is-primary is-large"
        name="chevron-right-icon"
      />
    </btn-stub>
  </div>
   
  <nav>
    <btn-stub
      class="has-dropshadow today-button"
      name="fade"
    >
      <icon-stub
        class="is-primary"
        name="rotate-ccw-icon"
      />
    </btn-stub>
     
    <marshmallow-select-stub
      arialabel="ariaLabels.selectPeriod"
      value="day.label"
    >
      <option
        value="0"
      >
        
        day.label
      
      </option>
      <option
        value="1"
      >
        
        week.label
      
      </option>
      <option
        value="2"
      >
        
        month.label
      
      </option>
      <option
        value="3"
      >
        
        year.label
      
      </option>
    </marshmallow-select-stub>
  </nav>
</content-header-stub>
`;
